tool/src/org/antlr/v4/codegen/model/Lexer.java

```diff
@@ -21,12 +22,11 @@
 	public final Collection<String> modes;
 	public final Collection<String> escapedModeNames;
 
-	@ModelElement public LinkedHashMap<Rule, RuleActionFunction> actionFuncs =
-		new LinkedHashMap<Rule, RuleActionFunction>();
+	@ModelElement public LinkedHashMap<Rule, RuleActionFunction> actionFuncs = new LinkedHashMap<>();
 
-	public Lexer(OutputModelFactory factory, LexerFile file) {
-		super(factory);
-		this.file = file; // who contains us?
```

Comment:
seems like that's a useful comment

Reply:
Maybe renamed `file` to `containingFile` instead of using such a comment?